gunicorn/workers/sync.py

```diff
@@ -186,6 +186,17 @@ def handle_request(self, listener, req, client, addr):
             finally:
                 if hasattr(respiter, "close"):
                     respiter.close()
+            # Discard data in the body before closing the socket to prevent
```

Comment:
This appearing in `sync.py` makes be think that the same code block needs to appear in `base_async.py` for the async workers. (It seems that there are essentially two copies of `handle_request` at this point, one in each of those files, that are almost the same.)